server/game/core/event/EventWindow.ts

```diff
@@ -0,0 +1,265 @@
+import { AbilityContext } from '../ability/AbilityContext';
+import CardAbilityStep from '../ability/CardAbilityStep';
+import { AbilityType } from '../Constants';
+import { TriggeredAbilityWindow } from '../gameSteps/abilityWindow/TriggeredAbilityWindow';
+import { BaseStepWithPipeline } from '../gameSteps/BaseStepWithPipeline';
+import { SimpleStep } from '../gameSteps/SimpleStep';
+import * as Contract from '../utils/Contract';
+
+export enum TriggerHandlingMode {
+
+    /** Any abilities triggered during this event window will not be resolved immediately but passed to the parent window */
+    PassesTriggersToParentWindow = 'passesTriggersToParentWindow',
+
+    /**
+     * Any abilities triggered during this event window or passed up from child windows will be resolved immediately.
+     * Typically `Auto` should be used instead.
```

Comment:
What do you mean by "Auto" here?

Reply:
That was something I added and then removed in the PR, good catch